src/pipx/commands/install.py

```diff
@@ -33,6 +33,7 @@ def install(
     """Returns pipx exit code."""
     # package_spec is anything pip-installable, including package_name, vcs spec,
     #   zip file, or tar.gz file.
+    python_flag_was_passed = python != DEFAULT_PYTHON
```

Comment:
This, as discussed, is not really accurate which is the reason we evaluate `python_flag_passed` in `pipx/main`. You should get the information from there.

Reply:
Missed this commit. Updated. Also moved the `python_flag_passed=bool(args.python)` out of the if condition. Just to make it simpler to pass it to various methods if required.
```python3
    python_flag_passed = bool(args.python)
    if "python" in args:
        fetch_missing_python = args.fetch_missing_python
        try:
            interpreter = find_python_interpreter(
                args.python or DEFAULT_PYTHON, fetch_missing_python=fetch_missing_python
            )
            args.python = interpreter
```